tools/protovalidate-conformance/internal/cases/cases_any.go

```diff
@@ -117,5 +118,21 @@ func anySuite() suites.Suite {
 				},
 			),
 		},
+		"compilation/wrong_type/scalar": {
+			Message:  &cases.AnyWrongTypeScalar{Val: "test"},
+			Expected: results.CompilationError("mismatched message rules"),
```

Comment:
We're already using the term "message rules" for something else: https://buf.build/docs/protovalidate/schemas/standard-rules/#message-rules

"mismatched rule type and field type"?

(I'm aware compilation error messages aren't compared even with `--strict_message`, but it still seems off to use "message rules" here)